sys-boot/refind/refind-0.14.0.2-r1.ebuild

```
@@ -0,0 +1,183 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit toolchain-funcs secureboot
+
+DESCRIPTION="The UEFI Boot Manager by Rod Smith"
+HOMEPAGE="https://www.rodsbooks.com/refind/"
+SRC_URI="mirror://sourceforge/project/${PN}/${PV}/${PN}-src-${PV}.tar.gz"
+
+LICENSE="BSD GPL-2 GPL-3 FDL-1.3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+FS_USE="btrfs +ext2 +ext4 hfs +iso9660 ntfs reiserfs"
+IUSE="${FS_USE} custom-cflags doc"
+
+DEPEND="sys-boot/gnu-efi"
+
+# for ld.bfd and objcopy
+BDEPEND="sys-devel/binutils"
+
+DOCS=( README.txt NEWS.txt )
+
+PATCHES=( "${FILESDIR}"/${P}-clang.patch )
+
+checktools() {
+	if [[ ${MERGE_TYPE} != "binary" ]]; then
+		# 832018
+		export LD="$(tc-getLD)"
+		tc-ld-force-bfd
+		# tc-ld-force-bfd does not work in all cases, so try to also fix LD
+		LD="${LD/.lld/.bfd}"
+		tc-ld-is-lld "${LD}" && die "Linking with lld produces broken executables and may lead to unbootable system"
+
+		# 732256
+		# llvm-objcopy does not support EFI target, try to use binutils objcopy or fail
+		export OBJCOPY="$(tc-getOBJCOPY)"
+		OBJCOPY="${OBJCOPY/llvm-/}"
+		LANG=C LC_ALL=C "${OBJCOPY}" --help | grep -q '\<pei-' || die "${OBJCOPY} (objcopy) does not support EFI target"
+	fi
+}
+
+pkg_pretend() {
+	if use custom-cflags; then
+		ewarn
+		ewarn "You have enabled building with USE=custom-cflags. Be aware that"
+		ewarn "using this can result in EFI binaries that fail to run and may"
+		ewarn "fail to build at all. This is strongly advised against by upstream."
+		ewarn
+		ewarn "See https://bugs.gentoo.org/598587#c3 for more information"
+		ewarn
+	fi
+
+	checktools
+}
+
+pkg_setup() {
+	if use x86; then
+		export EFIARCH=ia32
+		export BUILDARCH=ia32
+	elif use amd64; then
+		export EFIARCH=x64
+		export BUILDARCH=x86_64
+	fi
+	secureboot_pkg_setup
+
+	# this does not only check, but also exports LD and OBJCOPY
+	checktools
+}
+
+src_prepare() {
+	default
+
+	# bug 598647 - PIE not supported
+	sed -e '/^CFLAGS/s/$/ -fno-PIE/' -i Make.common || die
+	sed -e '1 i\.NOTPARALLEL:' -i filesystems/Makefile || die
+
+	# 881131, 832018
+	if tc-is-clang; then
+		sed -e 's/-fno-tree-loop-distribute-patterns//' -i Make.common || die
+	fi
+
+	cp "${FILESDIR}"/refind-sbat-gentoo-${PV}.csv refind-sbat-gentoo.csv || die
+}
+
+src_compile() {
+	# Update fs targets depending on uses
+	local fs fs_names=()
+	for fs in ${FS_USE}; do
+		fs=${fs#+}
+		if use "${fs}"; then
+			fs_names+=(${fs})
```

We normally do e.g.
```suggestion
			fs_names+=( ${fs} )
```
but it's not strictly required.

---

I saw that, but this is a part I didn't touch.  Where I did changes I added spaces.  But I'll add them everywhere, looks nicer.